drivers/backup/controllers/cluster/cluster_namespace.go

```diff
@@ -0,0 +1,207 @@
+package cluster
+
+import (
+	"fmt"
+	"github.com/portworx/torpedo/drivers/backup/utils"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/portworx/torpedo/pkg/log"
+	"github.com/portworx/torpedo/tests"
+	"strings"
+	"time"
+)
+
+// ValidateSchedulerContextConfig holds the necessary configuration to effectively validate scheduler-context
+type ValidateSchedulerContextConfig struct {
+	waitForRunningTimeout       time.Duration
+	waitForRunningRetryInterval time.Duration
+	validateVolumeTimeout       time.Duration
+	validateVolumeRetryInterval time.Duration
+}
+
+// DestroySchedulerContextConfig holds the necessary configuration to effectively destroy scheduler-context
+type DestroySchedulerContextConfig struct {
+	waitForDestroy             bool
+	waitForResourceLeakCleanup bool
+	skipClusterScopedObjects   bool
+}
+
+// NamespaceConfig holds the necessary configuration for managing namespaces within a ClusterController
+type NamespaceConfig struct {
+	namespace  string
+	appKey     *string
+	contexts   []*scheduler.Context
+	controller *ClusterController
+	*ValidateSchedulerContextConfig
+	*DestroySchedulerContextConfig
+}
+
+// validateNamespaceConfig validates NamespaceConfig
+func validateNamespaceConfig(c *NamespaceConfig) error {
+	if c.contexts == nil {
+		err := fmt.Errorf("no scheduler-contexts of the namespace [%s] found", c.namespace)
+		return utils.ProcessError(err)
+	} else if len(c.contexts) == 0 {
+		if c.appKey == nil {
+			err := fmt.Errorf("no scheduler-contexts of the namespace [%s] found", c.namespace)
+			return utils.ProcessError(err)
+		} else {
+			err := fmt.Errorf("no scheduler-contexts of the namespace [%s] with the app-key [%s] found", c.namespace, *c.appKey)
+			return utils.ProcessError(err)
+		}
+	}
+	return nil
+}
+
+// Application filters the NamespaceConfig to include only the scheduler-contexts associated with the specified appKey
+func (c *NamespaceConfig) Application(appKey string) *NamespaceConfig {
+	if c.contexts == nil {
+		return c
+	}
+	c.appKey = &appKey
+	appContexts := make([]*scheduler.Context, 0)
+	for _, ctx := range c.contexts {
+		log.Infof("%s -- %s", ctx.App.Key, appKey+"-")
+		if strings.HasPrefix(ctx.App.Key, appKey+"-") {
+			appContexts = append(appContexts, ctx)
+		}
+	}
+	c.contexts = appContexts
+	return c
+}
+
+// Validate iterates through each scheduler-context in the NamespaceConfig and validates it
+func (c *NamespaceConfig) Validate() error {
+	err := validateNamespaceConfig(c)
+	if err != nil {
+		debugMessage := fmt.Sprintf("namespace-config: [%v]", c)
+		return utils.ProcessError(err, debugMessage)
+	}
+	err = c.controller.SwitchContext()
+	if err != nil {
+		return utils.ProcessError(err)
+	}
+	for _, ctx := range c.contexts {
+		log.Infof("Validating application [%s]", ctx.App.Key)
+		log.Infof("Waiting for application [%s] to start running", ctx.App.Key)
+		err = tests.Inst().S.WaitForRunning(ctx, c.waitForRunningTimeout, c.waitForRunningRetryInterval)
```

Comment:
is Inst() not accessible here? anything accessing from tests package might cause cycle dependency issues if not now in future

Reply:
I am avoiding the use of dot imports and instead explicitly stating the package name.